arelle/plugin/validate/ESEF/__init__.py

```diff
@@ -142,19 +192,46 @@ def validateXbrlFinally(val, *args, **kwargs):
     modelXbrl.profileActivity()
     modelXbrl.modelManager.showStatus(_statusMsg)
     
+    prefixedNamespaces = modelXbrl.prefixedNamespaces
+    
+    reportPackageMaxMB = val.authParam["reportPackageMaxMB"]
+    if reportPackageMaxMB is not None:
+        maxMB = float(reportPackageMaxMB)
+        if val.authParam["reportPackageMeasurement"] == "unzipped" and modelXbrl.fileSource.fs:
+            _size = sum(zi.file_size for zi in modelXbrl.fileSource.fs.infolist())
+        else:
+            _size = os.path.getsize(modelXbrl.fileSource.basefile)
+        if _size > maxMB * 1048576:
+            modelXbrl.error("arelle.ESEF.maximumReportPackageSize",
+                            _("The authority %(authority)s requires a report package size under %(maxSize)s MB, size is %(size)s."), 
+                            modelObject=modelXbrl, authority=val.authority, maxSize=reportPackageMaxMB, size=_size)
+                            
+    if val.authority == "UK":
+        if modelXbrl.fileSource and modelXbrl.fileSource.taxonomyPackage and modelXbrl.fileSource.taxonomyPackage["publisherCountry"] != "GB":
+            modelXbrl.error("UKFRC.1.2.publisherCountrySetting",
+                        _("The “Publisher Country” element of the report package metadata for a UKSEF report MUST be set to \"GB\" but was \"%(publisherCountry)s\"."), 
```

Comment:
The quotes around `Publisher Country` are unicode characters and need to be updated to ascii characters.